File: Terraform/Belcorp/Test/API4/modulos/lambda/code/handler.py
import os
import json
import boto3
from urllib.request import urlopen

translate = boto3.client('translate')

def lambda_handler(event, context):
    try:
        # Fetch the quote
        with urlopen("http://api.quotable.io/quotes/random") as response:
            quotes = json.loads(response.read())
            quote = quotes[0]

        # Translate the quote
        result = translate.translate_text(
            Text=quote['content'],
            SourceLanguageCode='en',
            TargetLanguageCode='es'
        )

        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'application/json'
            },
            'body': json.dumps({
                'cita': result['TranslatedText'],
                'autor': quote['author']
            })
        }
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({
                'error': str(e)
            })
        }

File: Terraform/Belcorp/Test/API3/modulos/lambda/code/handler.js
import { TranslateClient, TranslateTextCommand } from "@aws-sdk/client-translate";
const client = new TranslateClient({});

export const handler = async (event) => {
    // Fetch the quote
    const response = await fetch("http://api.quotable.io/quotes/random");
    const [quote] = await response.json();

    // Translate the quote
    const command = new TranslateTextCommand({
        Text: quote.content,
        SourceLanguageCode: "en",
        TargetLanguageCode: "es",
    });

    const { TranslatedText } = await client.send(command);

    return {
        statusCode: 200,
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            cita: TranslatedText,
            autor: quote.author,
        }),
    };
};